src/test/unit_tests/test_nc_health.js

```diff
@@ -601,19 +586,82 @@ mocha.describe('nsfs nc health', function() {
         mocha.it('Account with new_buckets_path missing and allow_bucket_creation true, invalid account', async function() {
             const account_invalid = { name: 'account_invalid', uid: 999, gid: 999, allow_bucket_creation: true };
             await exec_manage_cli(TYPES.ACCOUNT, ACTIONS.ADD, { config_root, ...account_invalid });
-            Health.get_service_state.restore();
-            Health.get_endpoint_response.restore();
             Health.all_account_details = true;
             Health.all_bucket_details = false;
-            const get_service_state = sinon.stub(Health, "get_service_state");
+            const get_service_state = sinon.stub(Health, 'get_service_state');
             get_service_state.onFirstCall().returns(Promise.resolve({ service_status: 'active', pid: 1000 }))
                 .onSecondCall().returns(Promise.resolve({ service_status: 'active', pid: 2000 }));
-            const get_endpoint_response = sinon.stub(Health, "get_endpoint_response");
+            const get_endpoint_response = sinon.stub(Health, 'get_endpoint_response');
             get_endpoint_response.onFirstCall().returns(Promise.resolve({response: {response_code: 'RUNNING', total_fork_count: 0}}));
             const health_status = await Health.nc_nsfs_health();
             assert.strictEqual(health_status.checks.accounts_status.valid_accounts.length, 1);
             assert.strictEqual(health_status.checks.accounts_status.invalid_accounts.length, 1);
             await exec_manage_cli(TYPES.ACCOUNT, ACTIONS.DELETE, { config_root, name: account_invalid.name });
         });
+
+        mocha.it('Health all condition - invalid config directory upgrade', async function() {
```

Comment:
Would you add a comment - what makes it invalid?

Reply:
invalid means that the upgrade failed, I rephrased the test's headline